values.go

```diff
@@ -133,10 +133,21 @@ func (value *Number) Equal(that Value) bool {
 }
 
 func (value *Number) String() string {
-	s := strconv.FormatInt(value.value, 10)
 	scale := value.typ.scale
 	if scale == 0 {
-		return s
+		return strconv.FormatInt(value.value, 10)
+	}
+
+	var (
+		s        string
+		negative bool
+	)
+
+	if value.value < 0 {
+		s = strconv.FormatInt(-value.value, 10)
+		negative = true
```

Comment:
minor: if you more the `buffer` declaration along with `s` (in the `var`) block, you can both instantiate `s` and `buffer.WriteRune('-')` in the same `if`